File: Java-Solutions/truncate-a-string.java
    public static String truncateString(String str, int num) {
        if ((str.length() - num) <= 0) {
            return str;
        } else if (num < 4) {
            return str.substring(0, num) + "...";
        } else {
            return str.substring(0, num - 3) + "...";
        }
    }

File: JavaScript-Solutions/truncate-a-string.js
function truncateString(str, num) {
  if ((str.length - num) <= 0) {
    return str;
  } else if (num < 4) {
    return str.slice(0, num) + "...";
  } else {
    return str.slice(0, num-3) + "...";
  }
}
